src/poly/multiopen.rs

```diff
@@ -382,138 +383,116 @@ mod tests {
+    };
+
+    use super::construct_intermediate_sets;
+    use crate::poly::Rotation;
+    use pasta_curves::{arithmetic::FieldExt, Fp};
+
+    use std::collections::BTreeMap;
+    use std::convert::TryFrom;
+
+    #[derive(Debug, Clone)]
+    struct MyQuery<F> {
+        point: F,
+        eval: F,
+        commitment: usize,
+    }
+
+    impl super::Query<Fp> for MyQuery<Fp> {
+        type Commitment = usize;
+        type Eval = Fp;
+
+        fn get_point(&self) -> Fp {
+            self.point
+        }
+
+        fn get_eval(&self) -> Self::Eval {
+            self.eval
+        }
 
-            assert_eq!(point_sets, new_point_sets);
+        fn get_commitment(&self) -> Self::Commitment {
+            self.commitment
         }
+    }
+
+    prop_compose! {
+        fn arb_point()(
+            bytes in vec(any::<u8>(), 64)
+        ) -> Fp {
+            Fp::from_bytes_wide(&<[u8; 64]>::try_from(bytes).unwrap())
+        }
+    }
 
-        let mut a = false;
-        let mut a_set = 0;
-        let mut b = false;
-        let mut b_set = 0;
-        let mut c = false;
-        let mut c_set = 0;
-        let mut d = false;
-        let mut d_set = 0;
-
-        for (i, mut point_set) in point_sets.into_iter().enumerate() {
-            point_set.sort();
-            if point_set.len() == 1 {
-                assert_eq!(point_set[0], points[4]);
-                assert!(!a);
-                a = true;
-                a_set = i;
-            } else if point_set.len() == 2 {
-                let mut v0 = [points[0], points[1]];
-                let mut v1 = [points[0], points[3]];
-                v0.sort();
-                v1.sort();
-
-                if &point_set[..] == &v0[..] {
-                    assert!(!b);
-                    b = true;
-                    b_set = i;
-                } else if &point_set[..] == &v1[..] {
-                    assert!(!c);
-                    c = true;
-                    c_set = i;
-                } else {
-                    panic!("unexpected");
-                }
-            } else if point_set.len() == 3 {
-                let mut v = [points[0], points[1], points[2]];
-                v.sort();
-                assert_eq!(&point_set[..], &v[..]);
-                assert!(!d);
-                d = true;
-                d_set = i;
-            } else {
-                panic!("unexpected");
+    prop_compose! {
+        fn arb_query(commitment: usize, point: Fp)(
+            eval in arb_point()
+        ) -> MyQuery<Fp> {
+            MyQuery {
+                point,
+                eval,
+                commitment
             }
         }
+    }
+
+    prop_compose! {
+        // List of point indices corresponding to each query index.
+        fn arb_queries_inner(num_points: usize, num_queries: usize)(
+            point_indices in vec(select((0..num_points).collect::<Vec<_>>()), num_queries)
+        ) -> Vec<usize> {
+            point_indices
+        }
```

Comment:
From PR gardening with @nuttycom and @therealyingtong: currently we end up with point indices being potentially duplicated (good, corresponds to matching rotations), but commitment indices being entirely unique (meaning we're only testing one rotation per column). This should return a list of tuples representing each query, so that we can also get duplicate commitment indices.